test/functionality/typing/constant-type-inference.test.ts

```diff
@@ -0,0 +1,33 @@
+import { beforeAll, describe, expect, test } from 'vitest';
+import { retrieveNormalizedAst, withShell } from '../_helper/shell';
+import type { NormalizedAst } from '../../../src/r-bridge/lang-4.x/ast/model/processing/decorate';
+import { RDataType, TypeInferencer } from '../../../src/typing/type-inferencer';
+
+describe.sequential('Infer types for constant expressions', withShell(shell => {
+	let logicalAst: NormalizedAst | undefined;
+	let numberAst: NormalizedAst | undefined;
+	let stringAst: NormalizedAst | undefined;
+	const inferencer = new TypeInferencer();
+
+	beforeAll(async() => {
+		logicalAst = await retrieveNormalizedAst(shell, 'TRUE');
+		numberAst  = await retrieveNormalizedAst(shell, '42');
+		stringAst  = await retrieveNormalizedAst(shell, '"Hello, world!"');
+
+	});
+
+	test(`Infer ${RDataType.Logical} for logical constants`, () => {
```

Comment:
If you want to keep this style for simple tests, maybe you want to use [`test.each`](https://vitest.dev/api/#test-each)?